cmd/soroban-cli/src/rpc/transaction.rs

```diff
@@ -0,0 +1,357 @@
+use soroban_env_host::xdr::{
+    ContractAuth, DiagnosticEvent, HostFunction, OperationBody, ReadXdr, SorobanTransactionData,
+    Transaction, TransactionExt, VecM,
+};
+
+use crate::rpc::{Error, LogEvents, SimulateTransactionResponse};
+
+// Apply the result of a simulateTransaction onto a transaction envelope, preparing it for
+// submission to the network.
+pub fn assemble(
+    raw: &Transaction,
+    simulation: &SimulateTransactionResponse,
+    log_events: Option<LogEvents>,
+) -> Result<Transaction, Error> {
+    let mut tx = raw.clone();
+
+    // Right now simulate.results is one-result-per-function, and assumes there is only one
+    // operation in the txn, so we need to enforce that here. I (Paul) think that is a bug
+    // in soroban-rpc.simulateTransaction design, and we should fix it there.
+    // TODO: We should to better handling so non-soroban txns can be a passthrough here.
```

Comment:
I am curious as to what's the usecase for multiple `InvokeHostFunction` operations in the same transaction (unless they can, maybe, be interspersed with operations of other types).

Reply:
One thing you can do is upload the wasm, create the contract, and invoke an initialization method in a single txn. You can also invoke functions on multiple contracts atomically (e.g. approve a token, do a deposit, disapprove the token).